.githooks/pre-commit

```diff
@@ -22,8 +21,40 @@ if [ "$BRANCH" = "development" ]; then
   exit 255
 fi
 
+# Function to check if any staged files have been modified
+check_modified_staged_files() {
+  local modified_files=$(git diff --name-only --cached)
+  for file in $modified_files; do
+    if [ -f "$file" ] && ! git diff --quiet "$file"; then
+      return 0  # Found a modified staged file
+    fi
+  done
+  return 1  # No modified staged files found
+}
+
 # For all other branches, run the pre-commit checks using lint-staged.
 if npm run check:staged; then
+  # Check if any staged files were modified during the pre-commit checks
+  if check_modified_staged_files; then
```

Comment:
This if should come before and be independent from the `if npm run check:staged" conditional. 

The sequence of events we want to address is:
- The user stages some changed files.
- The user tries to commit the staged changes
- The`npm run check:staged` fails and the hook exists
- The user makes changes to some of the files that were staged to address the issue(s).
- The user tries to commit again (but forgot to `git add` the changes).
  - This is when the check should find the files that were previously staged but which were not re staged after the new changes.